card/mod/core/set/all/fetch_helper.rb

```diff
@@ -1,38 +1,45 @@
-
 module ClassMethods
-
   # a fetch method to support the needs of the card controller.
   # should be in Decko?
   def controller_fetch args
-    opts = controller_fetch_opts args
+    card_opts = controller_fetch_opts args
     if args[:action] == "create"
       # FIXME: we currently need a "new" card to catch duplicates
       # (otherwise save will just act like a normal update)
       # We may need a "#create" instance method to handle this checking?
-      Card.new opts
+      Card.new card_opts
     else
-      mark = args[:id] || opts[:name]
-      Card.fetch mark, look_in_trash: args[:look_in_trash], new: opts
+      standard_controller_fetch args, card_opts
     end
   end
 
   private
 
+  def standard_controller_fetch args, card_opts
+    mark = args[:id] || card_opts[:name]
+    card = Card.fetch mark, look_in_trash: args[:look_in_trash], new: card_opts
+    card.assign_attributes card_opts if args[:assign] && card&.real?
```

Comment:
this is new API and should be discussed (because I don't love it!).  The issue is that I need to set the content of an existing card before I save it.  neither `card[content]` nor `add_item` works outside of the context of an update. this solution feels hacky.

Reply:
Do you need it in a view?

Reply:
yes, edit view.

the filter_items view helps you figure out which items you want, and then you go back to the edit view and see how it looks with those items in it (and reorder and delete items if necessary) before saving.

Reply:
can't we make `card[content]` work in this case?

Reply:
I'm using `card[content]`.  But it doesn't work alone; you also have to use `assign`